node/impl/full/eth.go

```diff
@@ -207,6 +208,56 @@ func (a *EthAPI) FilecoinAddressToEthAddress(ctx context.Context, filecoinAddres
 	return ethtypes.EthAddressFromFilecoinAddress(filecoinAddress)
 }
 
+func (a *EthAPI) FilecoinAddressToEthAddressV1(ctx context.Context, filecoinAddress address.Address, blkParam string) (ethtypes.EthAddress, error) {
+	// If the address is an "f0" or "f4" address, `EthAddressFromFilecoinAddress` will return the corresponding Ethereum address right away.
+	if eaddr, err := ethtypes.EthAddressFromFilecoinAddress(filecoinAddress); err == nil {
+		return eaddr, nil
+	} else if err != ethtypes.ErrInvalidAddress {
+		return ethtypes.EthAddress{}, xerrors.Errorf("error converting filecoin address to eth address: %w", err)
+	}
+
+	// We should only be dealing with "f1"/"f2"/"f3" addresses from here-on.
+	switch filecoinAddress.Protocol() {
+	case address.SECP256K1, address.Actor, address.BLS:
+		// Valid protocols
+	default:
+		// Ideally, this should never happen but is here for sanity checking.
+		return ethtypes.EthAddress{}, xerrors.Errorf("invalid filecoin address protocol: %s", filecoinAddress.String())
+	}
+
+	// Set default block parameter to "finalized" if not provided
+	if blkParam == "" || len(blkParam) == 0 {
+		blkParam = "finalized"
```

Comment:
@rvagg Going with "finalized" as the default here as we want the default to be the safest option here. 

The other thing we can do here is to get rid of the default and fail the request if client hasn't specified a `blkParam` for a f1/f2/f3 address.

Reply:
finalized seems fine to me, and _should_ be awesome post-f3.